addons/misra.py

```diff
@@ -1107,11 +1143,17 @@ def misra_13_6(data):
 
 def misra_14_1(data):
     for token in data.tokenlist:
-        if token.str != 'for':
-            continue
-        exprs = getForLoopExpressions(token)
-        if exprs and hasFloatComparison(exprs[1]):
-            reportError(token, 14, 1)
+        if token.str == 'for':
+            exprs = getForLoopExpressions(token)
+            if not exprs:
+                continue
+            for counter in findCounterTokens(exprs[1]):
+                if counter.valueType and counter.valueType.isFloat():
+                    reportError(token, 14, 1)
+        elif token.str == 'while':
+            exprs = isFloatCounterInWhileLoop(token)
```

Comment:
the name exprs is confusing to me. I would not use a variable I would just write:

    if isFloatCounterInWhileLoop(token):